pkg/metrics/metrics.go

```diff
@@ -96,15 +105,28 @@ func (metrics *metrics) CountResult(result api.Result) {
 }
 
 // ObserveDurationByState logs duration of the state
-func (metrics *metrics) ObserveDurationByState(state *api.StateItem) error {
+func (metrics *metrics) ObserveTotalDurationByState(state *api.StateItem) error {
 	if state.StartedAt.IsZero() {
 		return fmt.Errorf("cannot observe StateItem if StartedAt is not set")
 	}
 	duration := state.FinishedAt.Sub(state.StartedAt.Time)
 	if duration < 0 {
 		return fmt.Errorf("cannot observe StateItem if FinishedAt is before StartedAt")
 	}
-	metrics.Duration.With(prometheus.Labels{"state": string(state.State)}).Observe(duration.Seconds())
+	metrics.TotalDuration.With(prometheus.Labels{"state": string(state.State)}).Observe(duration.Seconds())
+	return nil
+}
+
+// ObserveDurationByState logs duration of the state
+func (metrics *metrics) ObserveCurrentDurationByState(state *api.PipelineStatus) error {
+	if state.StartedAt.IsZero() {
+		return fmt.Errorf("cannot observe StateItem if StartedAt is not set")
+	}
```

Comment:
You are not monitoring the process between create of the pr and the first state.

Reply:
I don't understand how this is resolved now. Could you explain?

Reply:
good point, thats why I had the state undefined in the logic but I throw it out during the review process. Will have a look on this again...

Reply:
If no state is set at all, count it with label `state="new"`, not `state=""`.